Lyrics-Plus continues to not work

## ℹ Computer information

- Spotify version: 1.1.68.628.geb44bd66
- Spicetify version: 2.6.3


## 📝 Provide detailed reproduction steps (if any)

1. load lyrics-plus page
2. something went wrong
3. reload page doesn't work

### ✔️ Expected result

_What is the expected result of the above steps?_
Page loads

### ❌ Actual result

_What is the actual result of the above steps?_
Page didn't load

## 📷 Screenshots

_Are there any useful screenshots?_
<img width="1440" alt="Screen Shot 2021-09-16 at 4 25 20 PM" src="https://user-images.githubusercontent.com/43358890/133700662-6e2dea4d-262b-4473-9a84-c06a3e5afec7.png">

The git version of spicetify throws this error in the dev console:
```
vendor~xpui.js:1 Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings. 
    at zs (vendor~xpui.js:29)
    at vendor~xpui.js:29
    at ra (vendor~xpui.js:29)
    at la (vendor~xpui.js:29)
    at aa (vendor~xpui.js:29)
    at oa (vendor~xpui.js:29)
    at xs (vendor~xpui.js:29)
    at ws (vendor~xpui.js:29)
    at Os (vendor~xpui.js:29)
    at vs (vendor~xpui.js:29)
    at vendor~xpui.js:29
    at t.unstable_runWithPriority (vendor~xpui.js:29)
    at hi (vendor~xpui.js:29)
    at yi (vendor~xpui.js:29)
    at vi (vendor~xpui.js:29)
    at bs (vendor~xpui.js:29)
    at Un (vendor~xpui.js:29)
    at HTMLDocument.r (vendor~xpui.js:1)
(anonymous) @ vendor~xpui.js:1
wa @ vendor~xpui.js:29
Ua.o.componentDidCatch.n.callback @ vendor~xpui.js:29
zi @ vendor~xpui.js:29
Hi @ vendor~xpui.js:29
Is @ vendor~xpui.js:29
t.unstable_runWithPriority @ vendor~xpui.js:29
hi @ vendor~xpui.js:29
Ps @ vendor~xpui.js:29
vs @ vendor~xpui.js:29
(anonymous) @ vendor~xpui.js:29
t.unstable_runWithPriority @ vendor~xpui.js:29
hi @ vendor~xpui.js:29
yi @ vendor~xpui.js:29
vi @ vendor~xpui.js:29
bs @ vendor~xpui.js:29
Un @ vendor~xpui.js:29
r @ vendor~xpui.js:1
vendor~xpui.js:1 

Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings. 
    at zs (vendor~xpui.js:29)
    at vendor~xpui.js:29
    at ra (vendor~xpui.js:29)
    at la (vendor~xpui.js:29)
    at aa (vendor~xpui.js:29)
    at oa (vendor~xpui.js:29)
    at xs (vendor~xpui.js:29)
    at ws (vendor~xpui.js:29)
    at Os (vendor~xpui.js:29)
    at vs (vendor~xpui.js:29)
    at vendor~xpui.js:29
    at t.unstable_runWithPriority (vendor~xpui.js:29)
    at hi (vendor~xpui.js:29)
    at yi (vendor~xpui.js:29)
    at vi (vendor~xpui.js:29)
    at bs (vendor~xpui.js:29)
    at Un (vendor~xpui.js:29)
    at HTMLDocument.r (vendor~xpui.js:1)
```

--- a/CustomApps/lyrics-plus/Pages.js
+++ b/CustomApps/lyrics-plus/Pages.js
@@ -40,8 +40,7 @@ const SyncedLyricsPage = react.memo(
         const [position, setPosition] = useState(0);
         useTrackPosition(() => {
             if (!Player.data.is_paused) {
-                const playback = Spicetify.Player.origin2.state.playback;
-                setPosition(playback.position_as_of_timestamp + (Date.now() - playback.timestamp));
+                setPosition(Spicetify.Player.getProgress());
             }
         });
 

--- a/Extensions/popupLyrics.js
+++ b/Extensions/popupLyrics.js
@@ -567,13 +567,14 @@ function PopupLyrics() {
             const { width, height } = ctx.canvas;
             ctx.imageSmoothingEnabled = false;
             ctx.save();
-            ctx.filter = `blur(${userConfigs.blurSize}px)`;
+            let blurSize = Number(userConfigs.blurSize);
+            ctx.filter = `blur(${blurSize}px)`;
             ctx.drawImage(
                 image,
-                -blur * 2,
-                -blur * 2 - (width - height) / 2,
-                width + 4 * blur,
-                width + 4 * blur
+                -blurSize * 2,
+                -blurSize * 2 - (width - height) / 2,
+                width + 4 * blurSize,
+                width + 4 * blurSize
             );
             ctx.restore();
             ctx.fillStyle = "#000000b0";

--- a/jsHelper/spicetifyWrapper.js
+++ b/jsHelper/spicetifyWrapper.js
@@ -155,7 +155,7 @@ const Spicetify = {
 // Wait for Spicetify.Player.origin and origin2 to be available
 // before adding following APIs
 (function waitOrigins() {
-    if (!Spicetify.Player.origin || !Spicetify.Player.origin2) {
+    if (!Spicetify.Player || !Spicetify.Player.origin || !Spicetify.Player.origin2 || !Spicetify.Player.origin._state) {
         setTimeout(waitOrigins, 10);
         return;
     }
@@ -179,11 +179,11 @@ const Spicetify = {
         }
     );
 
-    Spicetify.Player.origin2.state.addProgressListener((data) => {
+    setInterval(() => {
         const event = new Event("onprogress");
-        event.data = data.position;
+        event.data = Spicetify.Player.getProgress();
         Spicetify.Player.dispatchEvent(event);
-    });
+    }, 1);
 
     Spicetify.addToQueue = Spicetify.Player.origin2.player.addToQueue;
     Spicetify.removeFromQueue = Spicetify.Player.origin2.removeFromQueue;

--- a/src/preprocess/preprocess.go
+++ b/src/preprocess/preprocess.go
@@ -333,7 +333,7 @@ Spicetify.React.useEffect(() => {
 	// React Component: Context Menu and Right Click Menu
 	utils.Replace(
 		&input,
-		`=(\w+)=>(\w+\(\)\.createElement\(([\w\.]+),\(\w+,[\w\.]+\)\(\{\},\w+,\{action:"open",trigger:"right-click"\}\)\))`,
+		`=(\w+)=>(\w+\(\)\.createElement\(([\w\.]+),\w*\((\w+,[\w\.]+)?\)\(\{\},\w+,\{action:"open",trigger:"right-click"\}\)\))`,
 		`=Spicetify.ReactComponent.RightClickMenu=${1}=>${2};Spicetify.ReactComponent.ContextMenu=${3}`)
 
 	// React Component: Context Menu - Menu

